recipe_estimator.py

```diff
@@ -51,19 +76,23 @@ def add_to_relative_constraint(solver, relative_constraint, ingredient_numvar, c
                 add_to_relative_constraint(solver, child_constraint, child_numvar, 1.0)
                 add_to_relative_constraint(solver, child_constraint, child_numvars[i+1], -1.0)
     else:
-        #print(relative_constraint.name(), ingredient_numvar['ingredient']['text'], coefficient)
+        print("relative_constraint:", relative_constraint.name(), ingredient_numvar['ingredient']['text'], coefficient)
         relative_constraint.SetCoefficient(ingredient_numvar['numvar'], coefficient)
 
+# For each ingredient, get the quantity estimate from the solver (for leaf ingredients)
+# or sum the quantity estimates of the child ingredients (for non-leaf ingredients)
 def get_quantity_estimate(ingredient_numvars):
     total_quantity = 0
+    quantity_estimate = 0
```

Comment:
I'm not sure what these changes are trying to do but they have broken the unit tests around quantity_estimates and evaporation. The idea of the quantity_estimate is that it is the amount of original ingredient that is needed to make 100g of the product, i.e. the quantity before evaporation

Reply:
It's still the same definition of quantity_estimate, except that it adds quantity estimates for parent ingredients as well.